test/cookbooks/test/recipes/certificate.rb

```diff
@@ -11,51 +19,109 @@
+windows_certificate 'delete certificate by thumbprint with space' do
+  source 'b1 bc 96 8b d4 f4 9d 62 2a a8 9a 81 f2 15 01 52 a4 1d 82 9c'
+  action :delete
+end
+
+# Add (.CER) DER encoded binary X.509 certificate in MY certificate store
 windows_certificate 'C:/certs/der-cert1.cer' do
   action :create
 end
 
+# Add (.CER) Base64 encoded X.509 certificate in ROOT certificate store
 windows_certificate 'C:/certs/base64-cert2.cer' do
   action :create
+  store_name 'CA'
 end
 
-windows_certificate '2796bae63f1801e277261ba0d77770028f20eee4' do
-  action :delete
+# Add (.CRT) format certificate in MY certificate store
+windows_certificate 'add .crt certificate' do
+  action :create
+  source 'C:/certs/test_cert.crt'
 end
 
-# this resource should always be "up to date"
-windows_certificate 'duplicate delete' do
-  source '2796bae63f1801e277261ba0d77770028f20eee4'
+# delete certificate by thumbprint with colon from MY certificate store
+windows_certificate 'delete certificate by thumbprint with colon' do
+  source 'b1:bc:96:8b:d4:f4:9d:62:2a:a8:9a:81:f2:15:01:52:a4:1d:82:9c'
   action :delete
 end
 
-# Generate using:
-# makecert -r -n "CN=ChefDummyCertForTest" -pe -ss My -sv test-cert.pvk test-cert.cer
-# pvk2pfx -pvk test-cert.pvk -spc test-cert.cer -pfx test-cert.pfx -po chef123
+# Add (.PFX) format certificate with password in CA certificate store
 windows_certificate 'C:/certs/test-cert.pfx' do
   action :create
   pfx_password 'chef123'
   store_name 'CA'
 end
 
-windows_certificate_binding 'ChefDummyCertForTest' do
+# delete certificate by thumbprint from CA certificate store
+windows_certificate '50 81 f6 67 f1 ef 00 5d 0e c3 9f a3 e3 0a a7 1b 4f d8 4e b6' do
+  action :delete
   store_name 'CA'
-  port 443
 end
 
-windows_certificate_binding '444-appid' do
-  cert_name 'ChefDummyCertForTest'
-  store_name 'CA'
-  port 444
-  app_id '{00000000-0000-0000-0000-000000000000}'
+# (.P7B) format certificate in MY certificate store
+windows_certificate 'add .p7b certificate' do
+  action :create
+  source 'C:/certs/test_p7b.p7b'
+  store_name 'MY'
 end
 
-windows_certificate_binding '443-hostname' do
-  cert_name 'ChefDummyCertForTest'
+# delete certificate by thumbprint from MY certificate store
+windows_certificate 'delete certificate by thumbprint with space' do
+  source 'b1 bc 96 8b d4 f4 9d 62 2a a8 9a 81 f2 15 01 52 a4 1d 82 9c'
+  action :delete
+end
+
+windows_certificate 'C:/certs/test-cert.pfx' do
+  action :create
+  pfx_password 'chef123'
   store_name 'CA'
-  address 'www.chef.io'
 end
+
+# windows_certificate_binding 'ChefDummyCertForTest' do
+#   store_name 'CA'
+#   port 443
+# end
+
+# windows_certificate_binding '444-appid' do
+#   cert_name 'ChefDummyCertForTest'
+#   store_name 'CA'
+#   port 444
+#   app_id '{00000000-0000-0000-0000-000000000000}'
+# end
+
+# windows_certificate_binding '443-hostname' do
+#   cert_name 'ChefDummyCertForTest'
+#   store_name 'CA'
+#   address 'www.chef.io'
+# end
```

Comment:
We should delete this commented code if we do not intend to use it.

Reply:
sure @rhass